uni/lib/view/course_unit_info/course_unit_info.dart

```diff
@@ -106,6 +120,23 @@ class CourseUnitDetailPageViewState
     return CourseUnitSheetView(sheet);
   }
 
+  Widget _courseUnitFilesView(BuildContext context) {
+    final sheet = context
+        .read<CourseUnitsInfoProvider>()
+        .courseUnitsFiles[widget.courseUnit];
```

Comment:
`read` is not reactive. Should we use `watch` here instead? (It might be happening that files are getting loaded quickly enough at the first tab and already loaded when the user taps the third tab)